docs/v2-system-addon/data_events.md

```diff
@@ -195,3 +195,82 @@ Here are different scenarios that cause a session end event to be sent:
 3. Closing the browser
 5. Refreshing
 6. Navigating to a new URL via the url bar or file menu
+
+### Session performance data
+
+This data is held in a child object of the `activity_stream_session` event called `perf`.  All fields suffixed by `_ts` are type `DOMHighResTimeStamp` (aka a double of milliseconds, with a 5 microsecond precision) with 0 being the [timeOrigin](https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp#The_time_origin) of the browser's hidden chrome window.
+
+An example might look like this:
+
+```js
+{
+  // XXX fix up the numbers to be more believable examples.
+  //
+  // timestamp of the action perceived by the user to trigger the load of this
+  // page
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2658
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2539
+
+  "load_trigger_ts": 1,
+
+  // What was the perceived trigger of the load action:
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2658
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2539
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2685
+
+  "load_trigger_type": [
+    "first_window_open" | // home only
+    "subsequent_window_open" | // home only
+    "toolbar_button" | // home only
+    "menu_plus_or_keyboard " | // newtab only
+    "session_restore" | // home or newtab
+    "url_bar" | // home or newtab
+    "refresh" | // home or newtab
+    "other"], // home or newtab
+
+  // As of this writing, this will be false for the first tab in every window,
+  // and true for every subsequent tab in that window.
+  //
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2539
+  "new_tab_preloaded": false,
+
+  // when the page itself receives an event that document.visibilityStat=visible
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2539
+  "visibility_event_fired_ts": 2,
+
+  // Most likely thing (based on previous telemetry) that the user is going
+  // to interact with (i.e. Hero element).
+  //
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2661
+  "top_sites_painted_ts": 5,
+
+  // When the entire page has been painted (not including stuff like screenshots
+  // showing up later).
+  //
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2662
+  "display_done_ts": 7,
+
+  // XXX below here, things are more up in the air.  We may not want to use
+  // the schema proposed below, or even implement all of the stuff to be
+  // collected here.
+
+  // See when the data was ready in case if and only if it happened
+  // after the data was required to render.
+  //
+  // TO BE IMPLEMENTED: https://github.com/mozilla/activity-stream/issues/2672
+  "top_sites_data_ready_ts": 9,
+  "highlights_data_ready_ts": 10,
```

Comment:
Since "highlights" won't make it to the system addon, shall we just take it out here? For other feeds (bookmarks, pocket, visit again), we can add them in later.